### PR TITLE
Add OpenBSD support

### PR DESCRIPTION
**Pull request by postmodern**
* Add OpenBSD to the CI matrix.
* Workaround OpenBSD's `tar` not supporting the `-J` option by using `xzcat $file | tar xf - -C $dest`.

### CODE REVIEW

**Comment by postmodern** [comment]
I'm not sure `pkg_add ruby%3.1` is the correct syntax. Only example I could find is this one: https://github.com/search?q=pkg_add+ruby%253.1&type=code

All other examples show `pkg_add ruby-3.1.2`, etc.

**Comment by postmodern** [comment]
https://github.com/kolargol/openvpn/blob/58ff7985cbb4c75153650435393dc8886be07378/openvpn.yml#L14
Appears that `pkg_add python%2.7` is pretty common, so it *should* work.

**Comment by flavorjones** [comment]
@postmodern Thank you for submitting this! I just kicked off CI.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -115,3 +115,33 @@ jobs:
             gem install bundler
             bundle install
             bundle exec rake ${{ matrix.task }}
+
+  openbsd:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: vmactions/openbsd-vm@v1
```
Comment (flavorjones) [inline .github/workflows/ci.yml:123]: Although I knew about `vmactions/freebsd-vm` I did not know there were other BSD variations available!

**Comment by flavorjones** [comment]
The macos failures are related to the macos-14 defaults that github started using, I'll fix those in a separate PR.

**Comment by flavorjones** [comment]
I've rebased onto `origin/main` to pick up #142 fixes.

**Comment by flavorjones** [comment]
Adding more packages and using gmake in a commit I just appended.

**Comment by flavorjones** [comment]
Added a commit that I think will fix the cmake tests ... still failing on some pkg-config edge cases we need to look into.

**Comment by flavorjones** [comment]
Really struggling here ... and the long feedback loops (because timeout tests are taking 2.5 minutes for each retry) are not making it easy. I'm going to try to reduce timeouts so that I can then have faster feedback loops to try to fix the remaining failing tests.

**Comment by flavorjones** [comment]
@postmodern OK, I'm timing out on this for now.

Things going on in this PR in CI:

- two failures related to pkg-config behavior (which might be actual failures since the OpenBSD pkg-config package is non-gnu?)
- failures in the CMake suite related to finding the compilers (see also my comment at https://github.com/flavorjones/mini_portile/pull/130#issuecomment-2093542965)
- the network timeouts are really bad for some tests -- on the order of 2.5 minutes for each of 3 retries -- and I'm not sure why. This is making solving the above problems harder for me. I tried setting kernel parameter `net.inet.tcp.keepinittime` but that didn't do anything. I'm starting to wonder if Ruby's OpenURI class support for timeouts doesn't work on OpenBSD.

**Comment by postmodern** [comment]
> Adding more packages and using gmake in a commit I just appended.

It might be a good idea to test against bsdmake, which is a bit more limited than GNU make.

> failures in the CMake suite related to finding the compilers

Are we sure `cc` is even available by default? Maybe we need to `pkg_add clang` or `pkg_add gcc`?

> the network timeouts are really bad for some tests -- on the order of 2.5 minutes for each of 3 retries -- and I'm not sure why.

I also get random `1 retrie(s) left for tar.gz (Failed to open TCP connection to localhost:36985 (Connection refused - connect(2) for "localhost" port 36985))` errors when running `rake test` locally on Fedora Linux.

**Comment by flavorjones** [comment]
> It might be a good idea to test against bsdmake

The issue I was addressing is that the libraries we're using as test subjects (e.g., libyaml) are _not_ compatible with BSD's make, so we need to work around that.

> Are we sure cc is even available by default

I chatted with Stan in https://github.com/flavorjones/mini_portile/pull/130 and I'm going to try to write a PR that uses RbConfig to determine the compilers, and this whole problem with the CMake detection may go away. I'll try to get to that this week.

> I also get random 1 retrie(s) left for tar.gz ...

Yes, the test suite intentionally exercises the methods when there are network failures. The problem is that on most platforms, the timeouts are fast and on openbsd for some reason they're very slow (2.5 minutes each x 3 retries).

**Comment by flavorjones** [comment]
I've rebased this branch onto the branch from #144 which I think will fix the CMake test failures.

**Comment by flavorjones** [comment]
Rebased once more, should be easier to see failures now.

**Comment by flavorjones** [comment]
Added a commit: `mkmf_config` no longer relies on features that openBSD's pkg-config doesn't support.
